saleor/shipping/zip_codes.py

```diff
@@ -0,0 +1,76 @@
+import re
+
+
+def group_values(pattern, *values):
+    result = []
+    for value in values:
+        try:
+            val = re.match(pattern, value)
+        except TypeError:
+            result.append(None)
+        else:
+            result.append(val.groups() if val else None)
+    return result
+
+
+def cast_tuple_index_to_type(index, target_type, *tuples):
+    """Cast tuple index to type.
+
+    Return list of tuples same as received but with index item casted to
+    tagret_type.
+    """
+    result = []
+    for _tuple in tuples:
+        to_result = []
+        try:
+            for i, entry in enumerate(_tuple):
+                to_result.append(entry if i != index else target_type(entry))
+        except TypeError:
+            pass
+        result.append(tuple(to_result))
+    return result
+
+
+def compare_values(code, start, end):
+    if not code or not start:
+        return False
+    if not end:
+        return start <= code
+    return start <= code <= end
+
+
+def check_uk_zip_code(code, start, end):
```

Comment:
It would be useful to have docstrings in these functions to describe how the algorithm works, maybe with simple examples of Irish/UK zip codes to illustrate which will be included in the range etc.